src/main/scala/is/hail/annotations/UnsafeRow.scala

```diff
@@ -166,9 +166,13 @@ object UnsafeRow {
   }
 
   private val tArrayInt32 = TArray(TInt32)
+  private val tArrayInt32Req = TArray(TInt32, elementsRequired = true)
 
-  def readArrayInt(region: MemoryBuffer, aoff: Long): Array[Int] = {
-    val t = tArrayInt32
+  def readArrayInt(region: MemoryBuffer, aoff: Long, elementsRequired: Boolean = false): Array[Int] = {
```

Comment:
This is used for AD and PX/PL, so elementsRequired = true.

Reply:
is that the only place it's supposed to be used? if so I'll just go ahead and change tArrayInt32 to be elementsRequired = true?

Reply:
Yes.  This is all temporary until TGenotype is ripped out.